cli/factory.rs

```diff
@@ -758,11 +782,17 @@ impl CliFactory {
     };
 
     Ok(CliMainWorkerFactory::new(
-      StorageKeyResolver::from_options(cli_options),
-      cli_options.sub_command().clone(),
-      npm_resolver.clone(),
-      node_resolver.clone(),
       self.blob_store().clone(),
+      if cli_options.code_cache_enabled() {
+        Some(self.code_cache()?.clone())
+      } else {
+        None
+      },
+      self.feature_checker()?.clone(),
+      self.fs().clone(),
+      maybe_file_watcher_communicator,
+      self.maybe_inspector_server()?.clone(),
+      cli_options.maybe_lockfile().cloned(),
```

Comment:
Is it necessary to shuffle all these options in this PR? Makes it quite a bit harder to figure out what is actually important that's changed

Reply:
I'll do it in a separate PR.

Reply:
Actually, it seems too small of a change. The only thing that changed here was I added the `self.permission_desc_parser()?.clone(),` argument.